cartesi-rollups/tutorials/javascript-wallet.md

```diff
@@ -44,205 +43,119 @@ Import essential functions and classes from external modules:
+    body: JSON.stringify({ payload: notice.payload }),
+  });
 }
-
 ```
 
-Finally, let’s update our handle_inspect function to return an account balance of an address.
+## Implementing withdrawals and transfers
 
 ```js
+let input = data.payload;
+let str = Buffer.from(input.substr(2), "hex").toString("utf8");
+let json = JSON.parse(str);
+
+if (json.method == "transfer") {
+  let notice = wallet.erc20_transfer(json.from, json.to, json.erc20, BigInt(json.amount));
+  await fetch(rollup_server + "/notice", {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({ payload: notice.payload }),
+  });
+} else if (json.method == "withdraw") {
+  try {
+    let voucher = wallet.erc20_withdraw(json.from, json.erc20, BigInt(json.amount));
+    await fetch(rollup_server + "/voucher", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ payload: voucher.payload, destination: voucher.destination }),
+    });
+  } catch (error) {
+    console.log("ERROR");
+    console.log(error);
+  }
+}
 
-async function handle_inspect(data) {
- console.log("Received inspect request data " + JSON.stringify(data));
- const url = hexToString(data.payload).split("/");
+```
 
-return router.process(url[0], url[1]); // balance/account
+Below are the sample payloads for the `transfer` and `withdraw` methods:
+
+```json
+// "transfer" method
+{
+    "method": "transfer",
+    "from": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+    "to": "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
+    "erc20": "0xae7f61eCf06C65405560166b259C54031428A9C4",
+    "amount": 5000000000000000000
+}
+
+// "withdraw" method
+{
+    "method": "withdraw",
+    "from": "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
+    "erc20": "0xae7f61eCf06C65405560166b259C54031428A9C4",
+    "amount": 3000000000000000000
 }
-```
 
-## Implementing withdrawals
-In the `handle_advance()` function, add an implementation to set the dApp address so that withdrawals can happen trustless.
-
-```js 
-// set dapp address
-   if (
-     msg_sender.toLowerCase() ===
-     dAppAddressRelayContract.toLowerCase()
-   ) {
-     rollup_address = payload;
-     router.set_rollup_address(rollup_address, "ether_withdraw");
-     router.set_rollup_address(rollup_address, "erc20_withdraw");
-     router.set_rollup_address(rollup_address, "erc721_withdraw");
-
-console.log("Setting DApp address");
-     return new Notice(
-       `DApp address set up successfully to ${rollup_address}`
-     );
-   }
 ```
 
-Now, we can add a general try-catch block to handle different types of withdrawals.
+## Method references
 
 ```js
-try {
-     const jsonpayload = JSON.parse(payloadStr);
-     console.log("payload is");
-     return router.process(jsonpayload.method, data);
-   } catch (e) {
-     return new Error_out(`failed to process command ${payloadStr} ${e}`);
-   }
+balance_get: (_account: Address) => Balance
```

Comment:
Idk about having these over here, but I don't feel too strongly about it either.